src/Signature/S3SignatureV4.php

```diff
@@ -94,6 +94,16 @@ public function presign(
                 $this->getPresignedPayload($request)
             );
         }
+
+        //Payload is unknown, checksum will cause requests to fail.
+        if ($request->getMethod() === 'PUT') {
```

Comment:
Is it guaranteed that every put request will contain an unknown payload?

Reply:
For presigned URLs, yes.  We cannot determine the payload because we are only generating a URL with some headers moved to query params.  The payload is not factored into this, so if a body is provided (or empty) during request serialization, you end up with checksum headers converted into query params in the URL.

Reply:
S3 `GET` requests are not affected by this given the nature of a `GET` request